thor/src/traits.rs

```diff
@@ -0,0 +1,40 @@
+use crate::{Message, Result};
+use bitcoin::{util::psbt::PartiallySignedTransaction, Address, Amount, Transaction};
+
+#[async_trait::async_trait]
+pub trait NewAddress {
+    async fn new_address(&self) -> Result<Address>;
+}
+
+#[async_trait::async_trait]
+pub trait BroadcastSignedTransaction {
+    async fn broadcast_signed_transaction(&self, transaction: Transaction) -> Result<()>;
+}
+
+/// Sign one of the inputs of the `FundingTransaction`.
+#[async_trait::async_trait]
+pub trait SignFundingPSBT {
+    async fn sign_funding_psbt(
+        &self,
+        psbt: PartiallySignedTransaction,
+    ) -> Result<PartiallySignedTransaction>;
+}
+
+#[async_trait::async_trait]
+pub trait BuildFundingPSBT {
+    async fn build_funding_psbt(
+        &self,
+        output_address: Address,
+        output_amount: Amount,
+    ) -> Result<PartiallySignedTransaction>;
+}
```

Comment:
🙃 These are part of the public API of the create protocol. I would expect them to be in that module.

Reply:
The `protocols::create` module is not public so this statement does not really hold.

Reply:
Well it's public to the rest of the crate, which is the consumer of the protocol. If we were to extract the lower level create protocol into a separate crate, these traits would have to go with it.